File: JAVApb5/src/main/java/concurs/domain/Inscriere.java
package concurs.domain;

import java.util.Objects;

public class Inscriere implements Entity<Long>{
    private  Long id;
    private  Participant participant;
    private Proba proba;

    public Inscriere(Long id, Participant participant, Proba proba) {
        this.id = id;
        this.participant = participant;
        this.proba = proba;
    }

    public Participant getParticipant() {
        return participant;
    }

    public void setParticipant(Participant participant) {
        this.participant = participant;
    }

    public Proba getProba() {
        return proba;
    }

    public void setProba(Proba proba) {
        this.proba = proba;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Inscriere inscriere = (Inscriere) o;
        return Objects.equals(id, inscriere.id) &&
                Objects.equals(participant, inscriere.participant) &&
                Objects.equals(proba, inscriere.proba);
    }

    @Override
    public String toString() {
        return "Inscriere{" +
                "id=" + id +
                ", participant=" + participant +
                ", proba=" + proba +
                '}';
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, participant, proba);
    }

    @Override
    public Long getId() {
        return null;
    }

    @Override
    public void setId(Long aLong) {

    }


}

File: CSharp/domain/Inscriere.cs
namespace CSharp.domain
{
    public class Inscriere:Entity<long>
    {
        public long Id { get; set; }
        public Participant Participant { get; set; }
        public Proba Proba { get; set; }

        public Inscriere(Participant participant, Proba proba)
        {
            Participant = participant;
            Proba = proba;
        }
        
        public override string ToString()
        {
            return string.Format("[Inscriere: Id={0}, idParticipant={1},idProba={2}]", Id, Participant,Proba);
        }

        public override bool Equals(object obj)
        {
            return base.Equals(obj);
        }

        public override int GetHashCode()
        {
            return base.GetHashCode();
        }
    }
}